core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -50,9 +50,11 @@
           TimingCollector timingCollector,
           Collection<StatementCustomizer> customizers,
           MappingRegistry mappingRegistry,
-          Foreman foreman)
+          Foreman foreman,
+          CollectorFactoryRegistry collectorFactoryRegistry)
```

Comment:
I wonder if we should start aggregating the types like `MappingRegistry`, `Foreman`, and `CollectorFactoryRegistry` -- they seem to all be passed around to largely the same places, and often when we copy one as a child we end up copying them all.